internal/provider/resource_monitor_resource.go

```diff
@@ -0,0 +1,807 @@
+	resp.Diagnostics.Append(req.State.Get(ctx, &resourceMonitorDataV0)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	name := resourceMonitorDataV0.Name
+	notifyUsers := resourceMonitorDataV0.NotifyUsers
+	creditQuota := resourceMonitorDataV0.CreditQuota
+	frequency := resourceMonitorDataV0.Frequency
+	startTimestamp := resourceMonitorDataV0.StartTimestamp
+	endTimestamp := resourceMonitorDataV0.EndTimestamp
+	suspendTrigger := resourceMonitorDataV0.SuspendTrigger
+	suspendTriggers := resourceMonitorDataV0.SuspendTriggers
+	if !suspendTriggers.IsNull() {
+		suspendTriggersElements := make([]types.Int64, 0, len(suspendTriggers.Elements()))
+		suspendTriggers.ElementsAs(ctx, &suspendTriggersElements, false)
+		if len(suspendTriggersElements) > 0 {
+			suspendTrigger = suspendTriggersElements[0]
+		}
+	}
+	suspendImmediateTrigger := resourceMonitorDataV0.SuspendImmediateTrigger
+	suspendImmediateTriggers := resourceMonitorDataV0.SuspendImmediateTriggers
+	if !suspendImmediateTriggers.IsNull() {
+		suspendImmediateTriggersElements := make([]types.Int64, 0, len(suspendImmediateTriggers.Elements()))
+		suspendImmediateTriggers.ElementsAs(ctx, &suspendImmediateTriggersElements, false)
+		if len(suspendImmediateTriggersElements) > 0 {
+			suspendImmediateTrigger = suspendImmediateTriggersElements[0]
+		}
+	}
+	notifyTriggers := resourceMonitorDataV0.NotifyTriggers
+
+	trigggers := make([]resourceMonitorTriggerModel, 0)
+	if !suspendTrigger.IsNull() {
+		trigggers = append(trigggers, resourceMonitorTriggerModel{
+			Threshold:     suspendTrigger,
+			TriggerAction: types.StringValue("SUSPEND"),
+		})
+	}
+	if !suspendImmediateTrigger.IsNull() {
+		trigggers = append(trigggers, resourceMonitorTriggerModel{
+			Threshold:     suspendImmediateTrigger,
+			TriggerAction: types.StringValue("SUSPEND_IMMEDIATE"),
+		})
+	}
+	if !notifyTriggers.IsNull() {
+		notifyTriggersElements := make([]types.Int64, 0, len(notifyTriggers.Elements()))
+		notifyTriggers.ElementsAs(ctx, &notifyTriggersElements, false)
+		for _, v := range notifyTriggersElements {
+			trigggers = append(trigggers, resourceMonitorTriggerModel{
+				Threshold:     v,
+				TriggerAction: types.StringValue("NOTIFY"),
+			})
+		}
+	}
+	triggersObjectType := types.ObjectType{}.WithAttributeTypes(map[string]attr.Type{
+		"threshold":      types.Int64Type,
+		"trigger_action": types.StringType,
+	})
+	triggersSet, diags := types.SetValueFrom(ctx, triggersObjectType, trigggers)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resourceMonitorV1 := &resourceMonitorModelV1{
+		Name:           name,
+		NotifyUsers:    notifyUsers,
+		CreditQuota:    creditQuota,
+		Frequency:      frequency,
+		StartTimestamp: startTimestamp,
+		EndTimestamp:   endTimestamp,
+		Triggers:       triggersSet,
+		Id:             resourceMonitorDataV0.Id,
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, resourceMonitorV1)...)
+}
+
+type resourceMonitorModelV1 struct {
```

Comment:
Later it would be nice to split into different version-prefixed files. (nit-picking)